charmcraft/commands/store/__init__.py

```diff
@@ -1373,11 +1373,11 @@ def fill_parser(self, parser):
 
     def run(self, parsed_args):
         """Run the command."""
-        charm_name = get_name_from_metadata()
+        charm_name = self.config.name or get_name_from_metadata()
```

Comment:
if `metadata.yaml` exists, should it have preference over `self.config.name`?

Reply:
It doesn't really matter — `self.config.name` gets the info from `charmcraft.yaml`, and loading the config gives an error if there's a name defined in both, so they're mutually exclusive.

Good question though!